autogtp/Game.cpp

```diff
@@ -196,7 +199,8 @@ void Game::move() {
 
 void Game::setMovesCount(int moves) {
     m_moveNum = moves;
-    m_blackToMove = (moves % 2) == 0;
+    const auto handicap_in_use = !m_engine.m_commands.filter("handicap").isEmpty();
+    m_blackToMove = (moves % 2) == (handicap_in_use ? 1 : 0);
```

Comment:
I think this should keep count of the handicap stones.
If White has 2 stones handicap the move 3rd is Black so `(moves % 2) = 1`, but if white has three stones handicap move 4th is Black so `(moves % 2) = 0`

Reply:
According to GTP the handicap stones don’t count as moves so it is simply if there is handicap the white goes first

Reply:
But then the game starts always at move 0 and black goes first anyway.

Reply:
I don't understand what you mean. From the [GTP spec](http://www.lysator.liu.se/~gunnar/gtp/gtp2-spec-draft2/gtp2-spec.html#SECTION00073000000000000000):
> The handicap stones are not included in the move history.

So as far as I can tell, white goes first if there is handicap. If I don't do this then I see that black gets the first move after the handicap stones are placed. The behaviour I have coded appears to work as I expect but do let me know if you observe otherwise.

Reply:
You are right Black has the handicap stones.
But looking at the formula there is something wrong: 
if the game start at move 1 then `moves` is `1` we have `(moves % 2)` equals to `1`  so with `handicap_in_use` true, `m_blackToMove` is also true.
But we just agreed to the opposite.

Reply:
Ok, I think I understand now. Thanks for explaining.
The game always starts at move 0 (as per the pre-existing code and because the handicap stones are not part of the move history), so if there is no handicap then black moves on even numbered turns but if there is handicap then black moves on odd numbered turns.